v2v/tests/src/libnbd/libnbd.py

```
@@ -95,6 +100,99 @@ def test_unsanitized_hostname():
+
+        # 1. Create source with small data (sparse)
+        process.run(f"truncate -s 100M {SRC_IMG}", shell=True)
+        process.run(f"echo 'HELLO' | dd of={SRC_IMG} bs=1 seek=1M conv=notrunc", shell=True)
+
+        # 2. Create destination that is FULLY ALLOCATED (non-sparse) zeroes
+        process.run(f"dd if=/dev/zero of={DEST_ZERO} bs=1M count=100", shell=True)
+        LOG.info(f"Initial Dest Usage: {get_disk_usage(DEST_ZERO)}MB (Should be 100MB)")
+
+        # 3. Run nbdcopy with --destination-is-zero
+        # This tells nbdcopy it can skip writing zeroes, effectively punching holes
+        process.run(f"nbdcopy --destination-is-zero {SRC_IMG} {DEST_ZERO} -v", shell=True)
```

Ruff (0.15.9)

[error] Function call with `shell=True` parameter identified, security issue

(S604)